Gems/ROS2/Code/Source/Spawner/ROS2SpawnerInterface.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2SpawnerInterface.h"
+#include "ROS2SpawnPointComponent.h"
+#include "ROS2SpawnerComponent.h"
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/ComponentApplicationBus.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Component/EntityId.h>
+#include <AzCore/Component/TransformBus.h>
+#include <AzCore/Debug/Trace.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/RTTI/RTTIMacros.h>
+#include <AzCore/RTTI/TypeInfoSimple.h>
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <AzCore/std/utils.h>
+#include <AzToolsFramework/API/ToolsApplicationAPI.h>
+#include <Entity/EditorEntityInfoBus.h>
+
+namespace ROS2
+{
+    const AZ::Transform& ROS2SpawnerInterface::GetDefaultSpawnPose() const
+    {
+        return m_defaultSpawnPose;
+    };
+
+    const AZStd::vector<AZStd::pair<AZStd::string, AZStd::shared_ptr<AZ::Transform>>> ROS2SpawnerInterface::GetAllSpawnPoints() const
+    {
+        AZStd::vector<AZStd::pair<AZStd::string, AZStd::shared_ptr<AZ::Transform>>> allSpawnPoints;
+
+        AZ::EntityId levelEntityId;
+        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+            levelEntityId, &AzToolsFramework::ToolsApplicationRequests::GetCurrentLevelEntityId);
```

Comment:
It looks like you want access to gather all the entities from the level that is loaded.

The ToolsApplication is not needed for that case.

The EntityContextEventBBus [OnEntityContextLoadedFromStream](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Entity/EntityContextBus.h#L174-L179) notification EBus can be handled to get the list of game entities.

Reply:
Alternatively, you can